### ISSUE TITLE
Cant get a remote connection (Node-red/HomeAssistant) to work.

### ISSUE THREAD

**Issue by YOYONL89** [owfs/owfs#95]
Hello all,

Probally it has something todo with my configuration but i cant seem to get a connection to a remote machine running OWFS.
I tried to connect to xxx.xxx.xxx.xxxx:4304 but it gets timed out for some reason.
The HTTPD serverice at port 2121 is running allright and i can see my devices there.

What do i miss?

[config.txt](https://github.com/owfs/owfs/files/7061537/config.txt)

**Comment by YOYONL89** [comment]
Hello Miccoli,

Here the results of the commands:
Localmachine:
sudo ss -pa sport :4304
`Netid   State    Recv-Q   Send-Q     Local Address:Port     Peer Address:Port
tcp     LISTEN   0        128                    *:4304                *:*       users:(("owserver",pid=26848,fd=3),("systemd",pid=1,fd=128))`

owdir
`owdir: command not found`

Remote machine:
nc -zv xxx.xxx.xxx.xxx 4304
`nc: xxx.xxx.xxx.xxx (xxx.xxx.xxx.xxx:4304): Host is unreachable`

owdir -s xxx.xxx.xxx.xxx
`owdir: command not found`

**Comment by miccoli** [comment]
To use `owdir` you need to install `owshell` which is a CLI owserver client, very handy to debug configuration problems. 

From your report it seems that `owserver` is up and running on the local machine and accepting remote connections. On the contrary, the server machine is not reachable from the remote client. What happens if you `ping` the owserver?

This seem more like a routing/firewall problem, unrelated to the `owserver` configuration, which looks OK.

**Comment by YOYONL89** [comment]
Hi Miccoli,

Sorry for the small delay but i wasnt home for a short hollyday.
Pinging seems to fail as well which is a bit akward since both machines operate in the same subnet.
Nevertheless there might be an option to forward the bus directly to the container.
But in order to do so i need to know what i have to forward.
The USB Busmaster isnt reporting as a TTYUSBX so i cant use that, is there anything which can point me in that direction then?

**Comment by miccoli** [comment]
> The USB Busmaster isnt reporting as a TTYUSBX so i cant use that, is there anything which can point me in that direction then?

Which busmaster are you using? If it is a DS9490 then it is accessed directly via libusb: typically accessing `/dev/bus/usb/<bus number>/<device number>`

`/dev/ttyUSBn` is for serial devices accessed via UART to USB bridge, say the LinkUSB.

**Comment by YOYONL89** [comment]
Yes its a DS9490 but it throws a heck of a lot numbers in both bus number as also device number.
I think i have to play with this, ill keep you posted.

**Comment by YOYONL89** [comment]
Ok i have found the bus number etc but now im lost because its still not working but im not sure its a OWFS related thing.

Because i have Homeassistant running as a container i'm a bit limited to what i can do.
Im now forwarding /dev/bus/usb/001/007 which is the adapter to the container with the same name.
When i add the so called integration i can choose between OWServer or Sysbus.
By selecting Sysbus it comes up with /sys/bus/w1/devices/ but i cant seem to forward my /dev/bus/usb/001/007 to that folder.

Any chance you might know how to get this working?

**Comment by miccoli** [comment]
Sorry but I have no idea how the Homeassistant container works...

**Comment by romeocontrol** [comment]
Probably out of date …
why not use the add-on hassio-owserver.
Works very well with serial devices
RT
